### ISSUE TITLE
Rule proposal: `no-return-array-push`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1487]
`Array#push()` returns the new length of the array, most time `return array.push(foo)` is a mistake.

## Fail

```js
function foo() {
	return array.push(bar);
}
```

## Pass

```js
// If mean to return the length
function foo() {
	return array.length + bar.length;
}
```

```js
// If mean to push and exit
function foo() {
  if (true) {
    array.push(bar);
    return;
  }

  // ...
}
```

```js
// Have to use `eslint-disable`
function foo() {
    // eslint-disable-next-line unicorn/no-return-array-push
	return array.push(bar);
}
```

**Comment by sindresorhus** [comment]
This is now accepted.
